src/components/strategies/create/useDuplicateStrategy.ts

```diff
@@ -1,91 +1,96 @@
 
-export const useDuplicateStrategy = () => {
-  const navigate = useNavigate();
-  const search: StrategyCreateSearch = useSearch({ strict: false });
-  const { strategy: urlStrategy } = search;
-
-  const duplicate = (strategy: Partial<Strategy>) => {
-    const encodedStrategy = btoa(JSON.stringify(strategy));
+export const getDuplicateStrategyParams = (strategy: Strategy) => {
+  // Remove balances if needed
+  if (isOverlappingStrategy(strategy)) {
+    strategy.order0.balance = '0';
+    strategy.order1.balance = '0';
+  }
+  if (isEmptyOrder(strategy.order0)) strategy.order0.balance = '0';
+  if (isEmptyOrder(strategy.order1)) strategy.order1.balance = '0';
 
-    navigate({
-      to: '/strategies/create',
-      search: {
-        ...search,
-        strategy: encodedStrategy,
-      },
-    });
+  // initialize params
+  const { order0, order1, base, quote } = strategy;
+  const searchParams: StrategyCreateSearch = {
+    base: base.address,
+    quote: quote.address,
+    buyMin: roundSearchParam(order0.startRate),
+    buyMax: roundSearchParam(order0.endRate),
+    buyBudget: roundSearchParam(order0.balance),
+    sellMin: roundSearchParam(order1.startRate),
+    sellMax: roundSearchParam(order1.endRate),
+    sellBudget: roundSearchParam(order1.balance),
   };
+  for (const key in searchParams) {
+    const keyString = key as keyof StrategyCreateSearch;
+    if (!searchParams[keyString]) delete searchParams[keyString];
+  }
 
-  const decoded = decodeStrategyAndValidate(urlStrategy);
-  if (decoded) {
-    // marginal price should be calculated based on prices
-    if (decoded.order0.marginalRate) decoded.order0.marginalRate = '';
-    if (decoded.order1.marginalRate) decoded.order1.marginalRate = '';
-
-    // Remove balance if overlapping strategy because market price changed
-    if (isOverlappingStrategy(decoded)) {
-      decoded.order0.balance = '';
-      decoded.order1.balance = '';
-    }
-
-    // Clear order balance for opposite direction in disposable
-    if (decoded.strategyType === 'disposable') {
-      if (decoded.strategyDirection === 'buy') decoded.order1.balance = '';
-      if (decoded.strategyDirection === 'sell') decoded.order0.balance = '';
-    }
+  const isOverlapping = isOverlappingStrategy({ order0, order1 });
+  const isRecurring = !isEmptyOrder(order0) && !isEmptyOrder(order1);
+  if (isOverlapping) {
+    searchParams.strategySettings = 'overlapping';
+  } else if (isRecurring) {
+    const isLimit = isLimitOrder(order0) && isLimitOrder(order1);
+    searchParams.strategyType = 'recurring';
+    searchParams.strategySettings = isLimit ? 'limit' : 'range';
+  } else {
+    const isLimit = isLimitOrder(order0) || isLimitOrder(order1);
+    searchParams.strategyType = 'disposable';
+    searchParams.strategySettings = isLimit ? 'limit' : 'range';
+    searchParams.strategyDirection = order1.endRate === '0' ? 'buy' : 'sell';
   }
+  return searchParams;
+};
+
+export const useDuplicateStrategy = () => {
```

Comment:
given that we now have all the input fields in the search params, I think we could remove this hook all together.

Reply:
We need it to initialize the state of the order in `useCreateOrder`. We'll be able to remove it once we bind the inputs with the query params as you're doing in the simulator